res/res_agi.c

```diff
@@ -1091,111 +1090,85 @@ static int add_silence(struct ast_channel *chan, struct ast_frame *f, struct ast
         long int ts_diff=0, f_ptime, last_seq=0;
 	int64_t gap_ms=0;
 	struct timeval s_tv = ast_channel_get_rec_start_time(chan);
+	unsigned int themssrc=0;
+	int ssrc_change=0;
+
+        /*! Check SSRC */
+        if ((f->themssrc != 0) && (f->themssrc != ast_channel_get_last_ssrc(chan, stream_no))){
```

Comment:
If we are calling the function  `ast_channel_get_last_ssrc` multiple times, we could consider using a local variable instead.